src/use-input.ts

```diff
@@ -40,9 +40,11 @@ export const useInput = <T extends BaseInput>(host: T) => {
 
 		useEffect(() => {
 			const onMouseDown = <T extends Event>(e: T) => {
+				const element = document.querySelector('#input[part="input"]') as HTMLInputElement | null;
 				if (
 					e.defaultPrevented ||
-					(e.target as HTMLElement).matches('input, textarea, label')
+					(e.target as HTMLElement).matches('input, textarea, label') ||
+					element?.disabled
```

Comment:
host?.disabled

Reply:
@megheaiulian `host`  cannot be used to check disabled state - it is of type T which cannot be casted to e.g. HTMLInputElement. I've solved that with taking the root's (Shadow Root of the host) element input and checking its disabled state.